openmc/mesh.py

```diff
@@ -1309,7 +1329,7 @@ def from_hdf5(cls, group: h5py.Group):
         mesh_id = int(group.name.split('/')[-1].lstrip('mesh '))
 
         # Read and assign mesh properties
-        mesh = cls(mesh_id)
+        mesh = cls(mesh_id=mesh_id)
```

Comment:
This is not going to work since `r_grid` and `z_grid` are now required arguments